rust/transaction-filter/src/filters/write_set_change_filter.rs

```diff
@@ -0,0 +1,306 @@
+                if let Some(address) = &self.address {
+                    if address != &wr.address {
+                        return false;
+                    }
+                }
+                if let Some(resource_type) = &self.resource_type {
+                    if !resource_type.is_allowed_opt(&wr.r#type) {
+                        return false;
+                    }
+                }
+            },
+        }
+        true
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ModuleChangeFilter {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub address: Option<String>,
+}
+pub enum ModuleChange<'a> {
+    DeleteModule(&'a DeleteModule),
+    WriteModule(&'a WriteModule),
+}
+
+impl Filterable<ModuleChange<'_>> for ModuleChangeFilter {
+    #[inline]
+    fn validate_state(&self) -> Result<(), Error> {
+        if self.address.is_none() {
+            return Err(Error::msg("At least one of address must be set"));
+        };
+        Ok(())
+    }
+
+    #[inline]
+    fn is_allowed(&self, item: &ModuleChange) -> bool {
+        if let Some(address) = &self.address {
+            return match &item {
+                ModuleChange::DeleteModule(dm) => address == &dm.address,
+                ModuleChange::WriteModule(wm) => address == &wm.address,
+            };
+        }
+        true
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct TableChangeFilter {
+    pub handle: Option<String>,
+    pub key: Option<String>,
+    pub key_type_str: Option<String>,
+}
+
+pub enum TableChange<'a> {
+    DeleteTableItem(&'a DeleteTableItem),
+    WriteTableItem(&'a WriteTableItem),
+}
+impl Filterable<TableChange<'_>> for TableChangeFilter {
+    #[inline]
+    fn validate_state(&self) -> Result<(), Error> {
+        if self.handle.is_none() && self.key.is_none() && self.key_type_str.is_none() {
+            return Err(Error::msg(
+                "At least one of handle, key, or key_type must be set",
+            ));
+        };
+        Ok(())
+    }
+
+    #[inline]
+    fn is_allowed(&self, item: &TableChange) -> bool {
+        match &item {
+            TableChange::DeleteTableItem(dti) => {
+                if let Some(handle) = &self.handle {
+                    return handle == &dti.handle;
+                }
+                if let Some(key_type) = &self.key_type_str {
+                    if !dti
+                        .data
+                        .as_ref()
+                        .map_or(false, |dtd| key_type == &dtd.key_type)
+                    {
+                        return false;
+                    }
+                }
+                if let Some(key) = &self.key {
+                    if !dti.data.as_ref().map_or(false, |dtd| key == &dtd.key) {
+                        return false;
+                    }
+                }
+            },
```

Comment:
Some filters `&&` all the fields and some filters `||`. We should keep it consistent.